hubverse_annotator/app.py

```diff
@@ -235,56 +236,50 @@ def reference_date_and_location_ui(
         st.session_state.locations_list = (
             loc_lookup.get_column("long_name").sort().to_list()
         )
-    if "location_selection" not in st.session_state:
-        st.session_state.location_selection = st.session_state.locations_list[
-            0
-        ]
+    if "current_loc_id" not in st.session_state:
+        st.session_state.current_loc_id = 0
 
     def go_to_prev_loc():
-        current_loc = st.session_state.locations_list.index(
-            st.session_state.location_selection
-        )
-        if current_loc > 0:
-            st.session_state.location_selection = (
-                st.session_state.locations_list[current_loc - 1]
-            )
+        st.session_state.current_loc_id -= 1
 
     def go_to_next_loc():
-        current_loc = st.session_state.locations_list.index(
-            st.session_state.location_selection
-        )
-        if current_loc < len(st.session_state.locations_list) - 1:
-            st.session_state.location_selection = (
-                st.session_state.locations_list[current_loc + 1]
-            )
+        st.session_state.current_loc_id += 1
 
     ref_dates = sorted(get_reference_dates(forecast_table), reverse=True)
-
     selected_ref_date = st.selectbox(
         "Reference Date",
         options=ref_dates,
         format_func=lambda d: d.strftime("%Y-%m-%d"),
         key="ref_date_selection",
     )
-    current_loc = st.session_state.locations_list.index(
-        st.session_state.location_selection
-    )
-    first_loc_is_selected = current_loc == 0
-    last_loc_is_selected = (
-        current_loc == len(st.session_state.locations_list) - 1
+    st.selectbox(
+        "Location",
+        options=list(range(len(st.session_state.locations_list))),
+        index=st.session_state.current_loc_id,
+        key="current_loc_id",
+        format_func=lambda i: st.session_state.locations_list[i],
     )
-    location_select_box, previous_button, next_button = st.columns([3, 1, 1])
-    with location_select_box:
-        st.selectbox(
-            "Location",
-            options=st.session_state.locations_list,
-            key="location_selection",
+    prev_col, next_col = st.columns([1, 1])
+    with prev_col:
+        st.button(
+            "⏮️",
+            disabled=(st.session_state.current_loc_id == 0),
+            on_click=go_to_prev_loc,
```

Comment:
Replace this with `st.session_state.current_loc_id -= 1` and remove the function definition?